src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -764,6 +766,56 @@ public ByteArrayInputStream getThumbnail(long datasetId, long imageId) {
 
   }
 
+  /**
+   * Associates instrument metadata with an existing image in the OMERO server.
+   * @param instrument
+   * @param imageId
+   */
+  public void associateInstrumentWithImage(Instrument instrument, Long imageId) {
+    if (!isConnected()) {
+      this.connect();
+    }
+    // 1. Check if an image with the id is in OMERO. Throw a runtime exception if not
+    ImageData imageData;
+    try {
+      BrowseFacility browseFacility = gateway.getFacility(BrowseFacility.class);
+      imageData = browseFacility.getImage(securityContext, imageId);
+    } catch (DSOutOfServiceException dsOutOfServiceException) {
+      throw new RuntimeException("Error while accessing omero service: broken connection, expired session or not logged in", dsOutOfServiceException);
```

Comment:
MAJOR NIT: I'd change the Error message here to make it more distinct from the execution exception case:
```suggestion
      throw new RuntimeException("Could not access omero service due to broken connection, expired session or not logged in", dsOutOfServiceException);
```

Reply:
same remarks as for the execution exception